### ISSUE TITLE
Schema and number of arrays unequal for array containing documents

### ISSUE THREAD

**Issue by lazargugleta** [mongodb-labs/mongo-arrow#250]
Hey,

When I insert a document with an array and try to read it back from a collection with a non-existing field in a schema, it breaks with a ValueError.

```python
import bson
import pyarrow
import pymongoarrow.api

from pymongo import MongoClient

client = MongoClient()

client.db.collection.insert_one(
    {'_id': bson.objectid.ObjectId("000000000000000000000013"),
     'list_field': [{'name': "Test1", 
                     'test': "Test2"}]}
              )

schema = pymongoarrow.api.Schema({
    '_id': bson.ObjectId,
    'list_field': [{
        'name': pyarrow.string(),
        'test': pyarrow.string(),
        'test_test': pyarrow.string(),  # does not exist in the database collection
        }],
})

df = pymongoarrow.api.aggregate_polars_all(client.db.collection, query, schema=schema)
```

I get the following traceback to a ValueError:
```

  File "/home/test.py", in test
    df = pymongoarrow.api.aggregate_polars_all(self._collection, query, schema=schema)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gugleta/.cache/pypoetry/virtualenvs/py3.12/lib/python3.12/site-packages/pymongoarrow/api.py", line 368, in aggregate_polars_all
    return _arrow_to_polars(aggregate_arrow_all(collection, pipeline, schema=schema, **kwargs))
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gugleta/.cache/pypoetry/virtualenvs/py3.12/lib/python3.12/site-packages/pymongoarrow/api.py", line 152, in aggregate_arrow_all
    return context.finish()
           ^^^^^^^^^^^^^^^^
  File "/home/gugleta/.cache/pypoetry/virtualenvs/py3.12/lib/python3.12/site-packages/pymongoarrow/context.py", line 52, in finish
    return Table.from_arrays(arrays=arrays, schema=self.schema.to_arrow())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pyarrow/table.pxi", line 4851, in pyarrow.lib.Table.from_arrays
  File "pyarrow/table.pxi", line 1603, in pyarrow.lib._sanitize_arrays
ValueError: Schema and number of arrays unequal
```

This happens in version 1.6.0 but not in previous 1.5.2.
The problem that this causes is when the data in the database comes differently compared to the currently defined schema, let's say a field is removed, it can cause this exact issue without expecting it or being able to prevent it.

**Comment by ShaneHarvey** [comment]
Thanks for reporting this, we are investigating here: https://jira.mongodb.org/browse/INTPYTHON-418

We'll report back soon.

**Comment by blink1073** [comment]
Fixed in the 1.6.2 release.

**Comment by lazargugleta** [comment]
Thank you so much!
